core/src/test/scala/unit/kafka/admin/ReassignPartitionsCommandTest.scala

```diff
@@ -52,48 +55,71 @@ class ReassignPartitionsCommandTest extends ZooKeeperTestHarness with Logging wi
     val assigner = new ReassignPartitionsCommand(null, null)
 
-    //Given partitions 0 & 1 moves from broker 100 -> 102
-    val existing = Map(TopicAndPartition("topic1",0) -> Seq(100, 101), TopicAndPartition("topic1",1) -> Seq(100, 101))
-    val proposed = Map(TopicAndPartition("topic1",0) -> Seq(101, 102), TopicAndPartition("topic1",1) -> Seq(101, 102))
+    //Given partitions 0 & 1 moves from broker 100 -> 102. Partition 2 does not move.
+    val existing = Map(TopicAndPartition("topic1", 0) -> Seq(100, 101), TopicAndPartition("topic1", 1) -> Seq(100, 101), control)
+    val proposed = Map(TopicAndPartition("topic1", 0) -> Seq(101, 102), TopicAndPartition("topic1", 1) -> Seq(101, 102), control)
 
-      //When
-    val moves = assigner.replicaMoves(existing, proposed)
+    // Then
+    val mock = new TestAdminUtils {
+      override def changeTopicConfig(zkUtils: ZkUtils, topic: String, configChange: Properties): Unit = {
+        assertEquals("0:102,1:102", configChange.get(FollowerThrottledReplicasListProp)) //Should only be follower-throttle the moving replica
+        assertEquals("0:100,0:101,1:100,1:101", configChange.get(LeaderThrottledReplicasListProp)) //Should leader-throttle all existing (pre move) replicas
+      }
+    }
 
-    //Then moving replicas should be throttled
-    assertEquals("0:100,0:102,1:100,1:102", moves.get("topic1").get)
+    //When
+    assigner.assignThrottledReplicas(existing, proposed, mock)
   }
 
   @Test
   def shouldFindMovingReplicasMultipleTopics() {
+    val control = TopicAndPartition("topic1", 1) -> Seq(100, 102)
     val assigner = new ReassignPartitionsCommand(null, null)
 
-    //Given partition 0 on topics 1 & 2 move from broker 100 -> 102
-    val existing = Map(TopicAndPartition("topic1",0) -> Seq(100, 101), TopicAndPartition("topic2",0) -> Seq(100, 101))
-    val proposed = Map(TopicAndPartition("topic1",0) -> Seq(101, 102), TopicAndPartition("topic2",0) -> Seq(101, 102))
+    //Given topics 1 -> move from broker 100 -> 102, topics 2 -> move from broker 101 -> 100
+    val existing = Map(TopicAndPartition("topic1", 0) -> Seq(100, 101), TopicAndPartition("topic2", 0) -> Seq(101, 102), control)
+    val proposed = Map(TopicAndPartition("topic1", 0) -> Seq(101, 102), TopicAndPartition("topic2", 0) -> Seq(100, 102), control)
+
+    //Then
+    val mock = new TestAdminUtils {
+      override def changeTopicConfig(zkUtils: ZkUtils, topic: String, configChange: Properties): Unit = {
+        topic match {
+          case "topic1" =>
+            assertEquals("0:102", configChange.get(FollowerThrottledReplicasListProp))
+            assertEquals("0:100,0:101", configChange.get(LeaderThrottledReplicasListProp))
+          case "topic2" =>
+            assertEquals("0:100", configChange.get(FollowerThrottledReplicasListProp))
+            assertEquals("0:101,0:102", configChange.get(LeaderThrottledReplicasListProp))
+          case _ => fail()
```

Comment:
Probably worth including a message like `fail("Unexpected topic $topic")` so that when someone updates the test with more topics, they get a useful message.